js/stats.js

```diff
@@ -1,173 +1,222 @@
-	var pool = _.range(upperBound);
-	return _.map(_.range(numElements), n => _.sample(pool) + 1);
-}
-
-function sSplitHelper(numGroups, numPictures, targetRating, pictures, sSplitFunc) {
-  if(!_.isArray(pictures)) {
-		throwInvalidInput("Received something other than an array for pictures argument.");
-	}		
-	if(!checkIfPictures(pictures)) {
-		throwInvalidInput("Pictures arguments are not properly formatted.");
-	}
-	var size = _.size(pictures);		
-	if(numGroups * numPictures > size){
-		throwInvalidInput("Can't split " + size + " total pictures into " 
-			+ numGroups + "groups of " + numPictures);
-	}
-  
-  return sSplitFunc(numGroups, numPictures, targetRating, pictures);
-}
-
-function avg(nums) {
-  return _.reduce(nums, (m, n) => m + n, 0) / _.size(nums);
-}
-
-function printStats(sorted, targetRating) {
-  var i = 0;
-  var dist = 0;
-  _.each(sorted, grouping => {
-    var cavg = avg(_.pluck(grouping, 'rating'));
-    console.log('###group ' + i++ + '###');
-    console.log('avg: ' + cavg);
-    console.log('ratings: ' + _.pluck(grouping, 'rating'));
-    dist += Math.abs(targetRating - cavg);
-  });
-  console.log('total distance from targetRating: ~' + dist.toFixed(1));
-}
-
-
-/**
- * numGroups: number of groups you want to split the dataset into
- * numPictures: number of pictures you want in each group
- * targetRating: ideal average of each group of pictures
- * dataSetSize: how many randomly generated 'pictures' you want to pull from
- * ratingUpperBound: defines range of ratings, from 1 to ratingUpperBound.
- */
-function testSSplit(numGroups, 
-                     numPictures, 
-                     targetRating, 
-                     dataSetSize, 
-                     ratingUpperBound) {
-
-  var pictures = _.map(getRandomIntArray(dataSetSize, ratingUpperBound), r => new Picture("name", r, "filePath"));
-
-  console.log("sorting by greedy");
-  console.time('time taken (greedy)');
-  var sorted1 = sSplitHelper(numGroups, numPictures, targetRating, pictures, sSplitGreedy);
-  console.timeEnd('time taken (greedy)');
-  printStats(sorted1, targetRating);
-
-  console.log();
-
-  console.log("sorting by round robin");
-  console.time('time taken (round robin)');
-  var sorted2 = sSplitHelper(numGroups, numPictures, targetRating, pictures, sSplitRR);
-  console.timeEnd('time taken (round robin)');
-    printStats(sorted2, targetRating);
-
-  console.log();
-
-  console.log("sorting by random");
-  console.time('time taken (random)');
-  var sorted3 = sSplitHelper(numGroups, numPictures, targetRating, pictures, sSplitRandom);
-  console.timeEnd('time taken (random)');
-    printStats(sorted3, targetRating);
-}
-
-
-/* update this function with parameters */
-testSSplit(4, 10, 8, 100, 10);
-
-      
+/*
+    Author: Tony J Huang
+    Group: Team Stimulus
+    Created At: 2/18/15
+
+    This is the stats module, affectionately and globally referred 
+    to as _$_. This module provides the solutions to all of your 
```

Comment:
Why are we using the _$_ namespace? Don't really like this because A) it resemble's Jquery's $, and B) its inconvenient to type.